src/rust/ide/src/controller/module.rs

```diff
@@ -114,12 +113,11 @@ impl Handle {
     /// This function wont check if the definition under id exists.
     pub async fn executed_graph_controller_unchecked
     (&self, id:dr::graph::Id, project:&controller::Project)
-    -> FallibleResult<controller::ExecutedGraph> {
-        let definition_name = id.crumbs.last().cloned().ok_or_else(|| InvalidGraphId(id.clone()))?;
-        let graph           = self.graph_controller_unchecked(id);
-        let path            = self.path.clone_ref();
-        let execution_ctx   = project.create_execution_context(path,definition_name).await?;
-        Ok(controller::ExecutedGraph::new(graph,execution_ctx))
+    -> FallibleResult<model::ExecutedGraph> {
+        let method        = self.method_pointer(&id)?;
```

Comment:
Why there is an API which requires reference to ID which (I assume) implements Copy? it should consume ID instead. Oh I just learned that Id is really a Crumb. I would love to change the name then - Id means for me that its just a number. Seeing it, Im just sure its Copy. Instead it should be named `graph::Crumb`